sys/include/xtimer/tick_conversion.h

```diff
@@ -29,7 +29,78 @@ extern "C" {
 #endif
 
 /* Some optimizations for common timer frequencies */
+/*
+ * Adaptation for frequencies multiple of 921600, usually used to match
+ * baud rates on serial communications
+ */
+#if (XTIMER_HZ % 576 == 0)
 #if (XTIMER_SHIFT != 0)
+#if (XTIMER_HZ % 576 != 0)
+#error XTIMER_HZ must be a multiple of 576 when using XTIMER_SHIFT
+#endif
+/* This is a common frequency for baud rate generation. We use the fact that
+ * the greatest common divisor between 921600 and 1000000 is 1600, so instead of
+ * multiplying by the fraction (921600 / 1000000), we will instead use
+ * (576 / 625), which reduces the truncation caused by the integer widths */
+#if (XTIMER_HZ > 921600ul)
+#if (XTIMER_HZ != (921600ul << XTIMER_SHIFT))
+#error XTIMER_HZ != (921600ul << XTIMER_SHIFT)
+#endif
+/* XTIMER_HZ is a power-of-two multiple of 921600 Hz */
+/* e.g. Waspmote-pro uses a 1843200 Hz timer */
+inline static uint32_t _xtimer_ticks_from_usec(uint32_t usec) {
+    return div_u32_by_625div576(usec) << XTIMER_SHIFT;
+}
+
+inline static uint64_t _xtimer_ticks_from_usec64(uint64_t usec) {
+    return div_u64_by_625div576(usec) << XTIMER_SHIFT;
+}
+
+inline static uint32_t _xtimer_usec_from_ticks(uint32_t ticks) {
+    return div_u32_by_576div625(ticks) >> XTIMER_SHIFT;
+}
+
+inline static uint64_t _xtimer_usec_from_ticks64(uint64_t ticks) {
+    return div_u64_by_576div625(ticks) >> XTIMER_SHIFT;
+}
+#else /* (XTIMER_HZ < 921600ul) */
+#if ((XTIMER_HZ << XTIMER_SHIFT) != 921600ul)
+#error (XTIMER_HZ << XTIMER_SHIFT) != 921600ul
+#endif
+inline static uint32_t _xtimer_ticks_from_usec(uint32_t usec) {
+    return div_u32_by_625div576(usec) >> XTIMER_SHIFT;
+}
+
+inline static uint64_t _xtimer_ticks_from_usec64(uint64_t usec) {
+    return div_u64_by_625div576(usec) >> XTIMER_SHIFT;
+}
+
+inline static uint32_t _xtimer_usec_from_ticks(uint32_t ticks) {
+    return div_u32_by_576div625(ticks) << XTIMER_SHIFT;
+}
+
+inline static uint64_t _xtimer_usec_from_ticks64(uint64_t ticks) {
+    return div_u64_by_576div625(ticks) << XTIMER_SHIFT;
+}
+#endif /* defined(XTIMER_SHIFT) && (XTIMER_SHIFT != 0) */
+#elif (XTIMER_HZ == 921600ul)
+inline static uint32_t _xtimer_ticks_from_usec(uint32_t usec) {
+    return div_u32_by_625div576(usec);
+}
+
+inline static uint64_t _xtimer_ticks_from_usec64(uint64_t usec) {
+    return div_u64_by_625div576(usec);
+}
+
+inline static uint32_t _xtimer_usec_from_ticks(uint32_t ticks) {
+    return div_u32_by_576div625(ticks);
+}
+
+inline static uint64_t _xtimer_usec_from_ticks64(uint64_t ticks) {
+    return div_u64_by_576div625(ticks);
+}
+#endif /* (XTIMER_HZ == 921600ul) */
+#elif (XTIMER_SHIFT != 0) /* (XTIMER_HZ % 576 == 0) */
```

Comment:
Yes I was trying to solve the issue. Actually I need a way on which we can detect, first, if we're using either 1MHz or 921600Hz clock based timer, then we need to know if the shifting is used, to finally choose the right functions.

The current code will check first if there's shifting activated, then if it's a 1MHz or 921600Hz based clock. The process is repeated to ensure that we can choose 1MHz clocks.